components/RegisterISCN.vue

```diff
@@ -0,0 +1,523 @@
+    ipfsHash: string;
+  }>;
+}
+
+const languageOptions = ref([
+  { label: 'English', value: 'en' },
+  { label: '中文', value: 'zh' }
+])
+
+const iscnData = ref({
+  type: 'book',
+  title: '',
+  description: '',
+  isbn: '',
+  publisher: '',
+  publicationDate: '',
+  author: {
+    name: '',
+    description: '',
+    url: ''
+  },
+  license: 'All rights reserved',
+  contentFingerprints: [{ url: '' }],
+  downloadableUrls: [{
+    url: '',
+    type: '',
+    fileName: ''
+  }],
+  language: '',
+  tags: [],
+  ipfsHash: [],
+  arweaveId: [],
+  coverUrl: ''
+})
+
+const balance = ref(new BigNumber(0))
+const iscnFee = ref(new BigNumber(0))
+const iscnGasFee = ref('0')
+const uploadStatus = ref('')
+const error = ref('')
+const combinedArweaveIdList = computed(() => iscnData.value.arweaveId)
+const emit = defineEmits(['handleSubmit', 'submit'])
+
+const totalFee = computed(() => {
+  return iscnFee.value || new BigNumber(0)
+})
+
+const formattedSameAsList = computed(() => {
+  return iscnData.value.downloadableUrls
+    .filter(download => download.fileName && download.url)
+    .map((download) => {
+      if (download.fileName && download.type) {
+        return `${download.url}?name=${download.fileName.replace(/\s+/g, '')}`
+      }
+      return ''
+    })
+    .filter(Boolean)
+})
+const payload = computed(() => ({
+  type: iscnData.value.type,
+  name: iscnData.value.title,
+  description: iscnData.value.description,
+  author: iscnData.value.author.name,
+  authorDescription: iscnData.value.author.description,
+  license: iscnData.value.license,
+  contentFingerprints: iscnData.value.contentFingerprints.map(f => f.url),
+  inLanguage: iscnData.value.language,
+  publisher: iscnData.value.publisher,
+  isbn: iscnData.value.isbn,
+  datePublished: iscnData.value.publicationDate
+    ? new Date(iscnData.value.publicationDate).toISOString()
+    : undefined,
+  url: iscnData.value.contentFingerprints[0]?.url || '',
+  tagsString: iscnData.value.tags?.join(', ') || '',
+  ipfsHash: iscnData.value.ipfsHash,
+  arweaveId: iscnData.value.arweaveId,
+  sameAs: formattedSameAsList.value,
+  thumbnailUrl: iscnData.value.coverUrl
+}))
+
+const isFormValid = computed(() => {
+  const requiredFields = {
+    title: !!iscnData.value.title,
+    description: !!iscnData.value.description,
+    authorName: !!iscnData.value.author.name,
+    contentUrl: !!iscnData.value.contentFingerprints.some(f => !!f.url)
+  }
+
+  error.value = Object.entries(requiredFields)
+    .find(([_, isValid]) => !isValid)?.[0]
+    ?.toUpperCase() || ''
+
+  return Object.values(requiredFields).every(Boolean)
+})
+
+onMounted(() => {
+  const initialData = initializeFromSessionStorage()
+  if (initialData) {
+    iscnData.value = initialData
+  }
+  calculateISCNFee()
+})
+
+const initializeFromSessionStorage = () => {
+  if (process.server) {
+    return null
+  }
+
+  try {
+    const storedData = window.sessionStorage.getItem('uploadFileData')
```

Comment:
In some privacy setting `sessionStorage` is not available, we might want to consider using pinia store if `storedData` is a JSON-able data, with `sessionStorage` as fallback

Reply:
[♻️ Move uploadFile state to Pinia store](https://github.com/likecoin/nft-book-press/pull/292/commits/1225b5175ab232352b7f1753282b6a56e74c1ee6)